app/bundles/PluginBundle/Bundle/PluginDatabase.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\PluginBundle\Bundle;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\ORM\Tools\SchemaTool;
+use Mautic\IntegrationsBundle\Migration\Engine;
+use Mautic\PluginBundle\Entity\Plugin;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+class PluginDatabase
+{
+    private readonly string $mauticDbPrefix;
+
+    public function __construct(
+        private readonly EntityManagerInterface $em,
+        private readonly Connection $connection,
+        #[Autowire(env: 'MAUTIC_TABLE_PREFIX')]
+        ?string $mauticDbPrefix,
+    ) {
+        $this->mauticDbPrefix = $mauticDbPrefix ?? '';
+    }
+
+    /**
+     * Install plugin schema based on Doctrine metadata.
+     *
+     * @psalm-param array<ClassMetadata> $metadata
```

Comment:
I'm wondering why psalm?
```suggestion
     * @param array<ClassMetadata> $metadata
```

Reply:
A copy from the base class. Didn't want to interfere there to support the phpdoc later.

Reply:
Apologies, I don't get the reason. Can you rephrase?

Reply:
NVM, mixed up with another change :)